components/ws-proxy/pkg/sshproxy/server.go

```diff
@@ -21,6 +24,10 @@ import (
 
 const GitpodUsername = "gitpod"
 
+var ErrWorkspaceNotFound = errors.New("not found workspace")
+var ErrAuthFailed = errors.New("auth failed")
```

Comment:
it seems to be very generic, i.e. does not add any useful information, we already track `auth` as a stage and no that it failed. It would be interesting to know whether password is wrong or any other reasons